cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -311,13 +316,64 @@ public void resourceModified(T resource) {
             startupTaskCompletion.complete(null);
             // Cancel the scheduled termination
             terminationFuture.cancel(false);
+            // Start the job monitor if we haven't already done so.
+            // This is a workaround for CDAP-19134 - sometimes the watch doesn't get triggered on job completion.
+            if (!wrappedFuture.isDone()) {
+              Future<?> future = monitorScheduler.submit(() -> {
+                boolean terminateController = false;
+                while (true) {
```

Comment:
Don't loop inside the scheduler. To loop, you should resubmit the task from inside the task itself.